### PR TITLE
Add redux to react components

### PR DESCRIPTION
**Pull request by sameedmohsin28**
In this pull request, I have
- Added the addBook reducer to the ADD BOOK BUTTON.
- Added the removeBook reducer to the remove button.

### CODE REVIEW

File: src/components/InputBook.js [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+  return (
+    <>
+      <h3>ADD NEW BOOK</h3>
+      <form>
+        <input type="text" placeholder="Book Title" required />
+        <input type="text" placeholder="Author" required />
+        <button type="submit" onClick={handleBookSubmit}>ADD BOOK</button>
+      </form>
+    </>
+  );
```
Comment (mikethreels) [inline src/components/InputBook.js]: - [ ] According to the requirements you should save the values from the input fields in a local state and update them using `onchange()`

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Sameed, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job implementing the Redux store

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/InputBook.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+  const [inputBookAuthor, setInputBookAuthor] = useState('');
+
   const dispatch = useDispatch();
 
   const handleBookSubmit = (e) => {
     e.preventDefault();
-    const inputBookName = e.target.previousElementSibling.previousElementSibling.value;
-    const inputBookAuthor = e.target.previousElementSibling.value;
+    setInputBookName(e.target.previousElementSibling.previousElementSibling.value);
+    setInputBookAuthor(e.target.previousElementSibling.value);
```
Comment (emmiiorji) [inline src/components/InputBook.js]: You have done great by adding local states as requested by the previous reviewer. Kindly ensure that the state is updated on change of the input fields and not on submit of the form.

You'll need to add onChange properties to the input fields.

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @sameedmohsin28, 

# Status: Changes Required ♻️

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

<div align='center'>

</div>

 ## Good Highlights✔️

- Redux is properly added to components


 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @sameedmohsin28 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - All changes requested have been implemented 🥇  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**